### PR TITLE
api v2 updates

### PR DESCRIPTION
**Pull request by jon571**

### CODE REVIEW

**Comment by Fuhji** [comment]
cypress tests are failing but as they are about to be updated my my pr, might be worth waiting to merge those in first

**Comment by jon571** [comment]
I noticed that but had a look around and it looked like they were failing before my changes.  I might be wrong though.

**Comment by Fuhji** [comment]
> I noticed that but had a look around and it looked like they were failing before my changes. I might be wrong though.

nothing that gets merged into Develop has failing tests, literally cant merge until it passes. I've merged the updated tests now so feel free to re-merge when ready and ill then have a peek at the rest

**Comment by Fuhji** [comment]
@jon571 needs a rebase still.... 👀

**Comment by jon571** [comment]
Installing cypress...

Yawwwwn!

**Comment by Fuhji** [comment]
> Installing cypress...
> 
> Yawwwwn!

time for a new laptop then!

**Comment by jon571** [comment]
Cypress tests still failing, but now it's only a11y compliance tests.

File: src/app/apiAndObjects/api/diet/getDietaryAvailability.ts [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
+import { DietDataSource } from '../../objects/dietDataSource';
 import { DataLevel } from '../../objects/enums/dataLevel.enum';
-import { DataSource } from '../../objects/dataSource';
 import { SubRegionDataItem } from '../../objects/subRegionDataItem';
 import { CacheableEndpoint } from '../../_lib_code/api/cacheableEndpoint.abstract';
 import { RequestMethod } from '../../_lib_code/api/requestMethod.enum';
 
-export class GetSubRegionData extends CacheableEndpoint<SubRegionDataItem, GetSubRegionDataParams> {
-
+// TODO: NEEDS SPLITTING INTO TWO DIFFERENT ENDPOINTS!!!!
```
Comment (Fuhji) [inline src/app/apiAndObjects/api/diet/getDietaryAvailability.ts:10]: four exclamation marks must be serious 👀

**Review by Fuhji — APPROVED** [review]
great work as always @jon571 ✋

**Pull request merged**